pkg/lib/cockpit/_internal/transport.js

```diff
@@ -59,288 +59,291 @@ function parse_channel(data) {
 }
 
 /* Private Transport class */
-function Transport() {
-    const self = this;
-    self.application = calculate_application();
+class Transport {
+    constructor() {
```

Comment:
This is a really weird class, though -- it consists entirely of a constructor which magically adds things to `self` (which also ought to be `this`). Not a strong opinion, but I'm leaning towards leaving it as-is: It avoids this big (and potentially risky) code churn, and stands out as "weird" more obviously, to incicate that this needs a thorough rewrite. 

Alternatively, if you are bored: Things like `self.close = function close(options) { ...}` should become a proper `close(options) { ... }` method.

Reply:
I agree.  This is follow-up work.  I wanted to do the automated transformation because I know that it's safe.  It was a precondition for switching to `EventEmitter` (which requires that you subclass).

The problem with converting those functions to methods is that you need to be careful about how they're called.  The way they're implemented, they use `self` and are safe to pass around as callbacks.  If we turn them into class methods then you always need to do the call directly like `transport.close()`, or do `transport.close.bind(transport)` before using them as a callback.

I *think* this should be safe (considering the limited users of this API), and it's probably even worth doing.

Reply:
Right, I didn't yet see the subsequent commits when I typed this. `self` is a hack, but I don't mind that much -- it's also easy enough to grep for. I'm a bit more worried about not having proper methods, but I don't know how difficult it would be to convert them (merely syntactical, or if that changes behaviour in a subtle way).

As I said, not a veto, just thinking aloud.

Reply:
I went ahead and did a few more commits in the direction of splitting this up properly.  Hopefully it doesn't break anything :)